src/js/kits/model-kit-select/kit-preview.js

```diff
@@ -0,0 +1,62 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+
+const DEFAULT_THUMB = '/static/images/thumb.png';
+
+const KitPreview = (props) => {
+    const { kit, selected, htmlName, onToggle } = props;
+
+    const [checked, setChecked] = useState(selected);
+    const [thumbImg, setThumbImg] = useState(kit.box_image.small || DEFAULT_THUMB);
+    const [errored, setErrored] = useState(false);
+
+    const onThumbError = () => {
+        if (errored) return;
+        setThumbImg(DEFAULT_THUMB);
+        setErrored(true);
+    };
+
+    const onChange = (event) => {
+        const _checked = event.target.checked;
+        setChecked(_checked);
+        onToggle(kit, _checked);
```

Comment:
Depending on what you want to do in `onToggle` you could remove the whole `checked` state here and use `selected` prop directly. This avoids setting redundant state var, which will need to be kept in sync with props.

Reply:
there's going to be backend-information for the edit function where kits are already pre-selected. Keeping your remark in mind when I'm implementing that, it might indeed be simplified and be 100% prop-driven instead.